my-vault/internal/storage/postgres.go

```diff
@@ -0,0 +1,122 @@
+package storage
+
+import (
+	"database/sql"
+	"encoding/json"
+	"fmt"
+
+	"github.com/bmasaiti/go-projects/my-vault/internal/domain"
+)
+
+type PostgresSecretRepo struct {
+	DB *sql.DB
+}
+
+func NewPostgresSecretRepo(db *sql.DB) *PostgresSecretRepo {
+	return &PostgresSecretRepo{DB: db}
+}
+
+func (repo *PostgresSecretRepo) PutNewSecret(secret domain.Secret) error {
+	query := `
+		INSERT INTO secrets (id, name, kv_data) 
+		VALUES ($1, $2, $3)
+	`
+	kvDataJSON, err := json.Marshal(secret.KVMap)
+	if err != nil {
+		return fmt.Errorf("failed to marshal kv_data: %w", err)
+	}
+
+	_, err = repo.DB.Exec(query, secret.Id, secret.Name, kvDataJSON)
+	if err != nil {
+		return fmt.Errorf("failed to insert secret: %w", err)
+	}
+
+	return nil
+}
+
+func (repo *PostgresSecretRepo) GetScretsById(secretId string) (domain.Secret, error) {
+	query := `
+		SELECT id, name, kv_data
+		FROM secrets 
+		WHERE id = $1
+	`
+	var secret domain.Secret
+
+	var kvDataJSON []byte
+
+	err := repo.DB.QueryRow(query, secretId).Scan(
+		&secret.Id,
+		&secret.Name,
+		&kvDataJSON,
+	)
+	if err == sql.ErrNoRows {
+		return domain.Secret{}, fmt.Errorf("secret with id %s not found", secretId)
```

Comment:
I think we want to return an ErrNotFound, right?
```suggestion
		return domain.Secret{}, fmt.Errorf("secret with id %s not found: %w", secretId, ErrNotFound)
```